src/sentry/api/utils.py

```diff
@@ -234,6 +234,35 @@ def get_date_range_from_stats_period(
     return start, end
 
 
+def clamp_date_range(
+    range: tuple[datetime.datetime, datetime.datetime], max_timedelta: datetime.timedelta
+) -> tuple[datetime.datetime, datetime.datetime]:
+    """
+    Accepts a date range and a maximum time delta. If the date range is shorter
+    than the max delta, returns the range as-is. If the date range is longer than the max delta, clamps the range range, anchoring to the end.
+
+    If any of the inputs are invalid (e.g., a negative range) returns the range
+     without modifying it.
+
+    :param range: A tuple of two `datetime.datetime` objects
+    :param max_timedelta: Maximum allowed range delta
+    :return: A tuple of two `datetime.datetime` objects
+    """
+
+    [start, end] = range
+    delta = end - start
+
+    # Ignore negative ranges
+    if delta <= datetime.timedelta(0):
+        return (start, end)
+
+    # Ignore negative max time deltas
+    if max_timedelta < datetime.timedelta(0):
+        return (start, end)
+
+    return (end - min(delta, max_timedelta), end)
```

Comment:
Nit: I think you can clean this up a little bit by checking `if delta < max_delta`
```suggestion
    # Ignore negative max time deltas
    if max_timedelta < datetime.timedelta(0):
        return (start, end)
        
    # Ignore if delta is within acceptable range
    if delta < max_delta:
        return (start, end)

    return (end - max_timedelta, end)
```